general/pionerv1_price.js

```diff
@@ -0,0 +1,177 @@
+require('dotenv').config();
+const { BN, toBaseUnit, Web3, axios } = MuonAppUtils;
+
+const PionerV1App = {
+    APP_NAME: 'pionerV1_oracle',
+    useTss: true,
+
+    onRequest: async function (request) {
+        let { method, data: { params = {} } } = request;
+        switch (method) {
+            case 'price':
+                const { asset1, asset2 } = params;
+                const prices = await this.fetchPrices(asset1, asset2);
+                const isMarketOpen = this.isValidPriceData(prices, /* appropriate config */);
+
+                return {
+                    asset1: this.convertToBytes32(asset1),
+                    asset2: this.convertToBytes32(asset2),
+                    pairBid: this.scaleUp(prices.pairBid).toString(),
+                    pairAsk: this.scaleUp(prices.pairAsk).toString(),
+                    confidence: this.scaleUp(prices.confidence).toString()
+
+                };
+        }
+    },
+
+    signParams: function (request, result) {
+        const signTime = result.oldestTimestamp ? result.oldestTimestamp : Math.floor(Date.now() / 1000);
```

Comment:
This function will run on all nodes. `Date.now()` may vary on different nodes. Use `request.data.timestamp` instead. It is the time when the request starts on the network.